core/payments/reservation/reservation_ledger.go

```diff
@@ -0,0 +1,117 @@
+package reservation
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/core"
+)
+
+// Tracks usage of a single account reservation
+//
+// This struct is goroutine safe.
+type ReservationLedger struct {
+	config ReservationLedgerConfig
```

Comment:
Any particular reason why you don't use a `*ReservationLedgerConfig`? Although copying a struct is usually not a big deal for tiny structs, I typically make a habit of always passing them around by reference. That way, I'm never surprised if I end up passing around a struct that is expensive to copy.

Reply:
@samlaf has expressed the opinion in the past that configs should *not* be pointers, so I was just following that.

I'm not super opinionated, and can't remember the precise reasoning. Sam, care to weigh in?

Reply:
Not 100% convinced, but not something worth blocking the PR over.

Lack of a language level support for const/immutable fields is easily in the top 5 of my golang gripes.